[Hieu] iP

trunghjieu2506 opened the pull request.

Commit:
Level-0

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
A-CodingStandard

Alaneel commented on src/main/java/Corgi.java:
```diff
@@ -0,0 +1,94 @@
+            else {
+                addTask(input);
+            }
+        }
```
Recommend to close Scanner at the end of program

Alaneel commented on src/main/java/Corgi.java:
```diff
@@ -0,0 +1,94 @@
+        String input;
+
+        while (true) {
+            input = scanner.nextLine();
```
Recommend to trim the input to handle extra spaces

Alaneel replied:
Better to have validation to avoid malicious inputs

Alaneel reviewed (commented):
LGTM

Commit:
Level-4

Commit:
A-CodeQuality

Change from String Concatenation to String Format for cleaner code

yx-tzzz commented on src/main/java/Corgi.java:
```diff
@@ -0,0 +1,94 @@
+        while (true) {
+            input = scanner.nextLine();
+            if (input.equalsIgnoreCase("bye")) {
+                ArrayList<String> exit_message = new ArrayList<>(1);
```
Recommend ArrayList variable name to be in camelCase as per coding standard  (https://se-education.org/guides/conventions/java/basic.html#layout)

yx-tzzz commented on src/main/java/Corgi.java:
```diff
@@ -0,0 +1,94 @@
+        if (taskIndex >= 0 && taskIndex <= tasks.size()) {
+            unmark_print.add("Okay. I 've marked this task as undone:");
+            unmark_print.add(tasks.get(taskIndex).getStatusIcon());
+            corgiPrint(unmark_print);
```
Good use of variable and method names

yx-tzzz commented on src/main/java/Corgi.java:
```diff
@@ -0,0 +1,94 @@
+                unmarkTask(input);
+            }
+            else {
+                addTask(input);
```
Perhaps switch statement would be more suitable here?

yx-tzzz replied:
Recommend changing the if-else statements form to
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
as per coding standard (https://se-education.org/guides/conventions/java/basic.html#layout)

Commit:
Create hashmap for commands
Handle two basic exceptions of InvalidCommandException and MissingArgumentException

Commit:
Handle invalid command format exceptions: NumberFormatException, IndexOutOfBoundsException

Commit:
Update file packages and IndexOutOfBoundException for markTask and unmarkTask methods

Commit:
Remove old import

trunghjieu2506 committed:
add removeTask method with usage: delete <taskId>

trunghjieu2506 committed:
delete redundant exception class MarkingOutOfBoundException.java

trunghjieu2506 committed:
Save and Load files using ObjectSerialize. Objects are stored in tasks.dat file.

trunghjieu2506 committed:
Save and Load files using ObjectSerialize. Objects are stored in tasks.dat file.

trunghjieu2506 committed:
Merging branch-Level-6 into master

trunghjieu2506 committed:
Resolve merge conflict in Task.java

trunghjieu2506 committed:
Refactor package name.
Add saveList() for removeTask method

trunghjieu2506 committed:
Add Exception Handling for removeTask methods.

Update to FilePath for tasks.dat file to run regardless of OS environment

aditichadha1310 commented on src/main/java/CorgiManager/CorgiManager.java:
```diff
@@ -0,0 +1,262 @@
+                    throw new InvalidCommandException("Command does not exist!!");
+                }
+            } catch (MissingArgumentException e) {
+                //System.err.println might be better
```
The comments should be for the reader. This comment seems to be for your own reference.

aditichadha1310 commented on src/main/java/CorgiManager/CorgiManager.java:
```diff
@@ -0,0 +1,262 @@
+            int idx = input.indexOf("/by");
+            String deadlineName = input.substring(0, idx).trim();
+            // split the input into two parts (by and ...)
+            String deadlineTime = input.substring(idx + 1).trim().split(" ", 2)[1];
```
Avoid usage of magic strings and magic numbers.

trunghjieu2506 committed:
Code Refactoring for more OOP

trunghjieu2506 committed:
Follow SLAP and move low-level details of parsing in EventCommand.execute to Parser

trunghjieu2506 committed:
Remove redundant imports

trunghjieu2506 committed:
Update deprecated ExitCommand.class

trunghjieu2506 committed:
Update deprecated ExitCommand.class

trunghjieu2506 committed:
Follow SLAP by moving low-level implementation from command classes to Parser

trunghjieu2506 committed:
Merge branch-A-MoreOOP back into master

trunghjieu2506 committed:
Add FindCommand

trunghjieu2506 committed:
Merge pull request #1 from trunghjieu2506/branch-Level-9

Add FindCommand

trunghjieu2506 committed:
Add JavaDoc and some minor code refactoring

trunghjieu2506 committed:
Merge branch 'branch-A-JavaDoc'

trunghjieu2506 committed:
Minor bug

trunghjieu2506 committed:
Update print message when loading file storage

trunghjieu2506 committed:
Update README.md

trunghjieu2506 committed:
Update README.md

trunghjieu2506 committed:
Update ReadMe.md

trunghjieu2506 committed:
Update ReadMe.md

trunghjieu2506 committed:
Update README.md

trunghjieu2506 committed:
Update README.md
